src/script_interface/Variant.hpp

```diff
@@ -29,6 +31,64 @@ enum class VariantType {
   VECTOR
 };
 
+namespace detail {
+template <typename T, typename = void> struct infer_type_helper {};
+
+template <> struct infer_type_helper<bool> {
+  static constexpr VariantType value{VariantType::BOOL};
+};
+
+template <> struct infer_type_helper<std::string> {
+  static constexpr VariantType value{VariantType::STRING};
+};
+
+template <typename T>
+struct infer_type_helper<
+    T, typename std::enable_if<std::is_integral<T>::value, void>::type> {
+  static constexpr VariantType value{VariantType::INT};
+};
+
+template <typename T>
+struct infer_type_helper<
+    T, typename std::enable_if<std::is_floating_point<T>::value, void>::type> {
+  static constexpr VariantType value{VariantType::DOUBLE};
+};
```

Comment:
This substitution allows up-coversion to `long double` which results in a bogus digits in the extra decimal places.  #1051 is a good example for this.